src/components/auth/Register.tsx

```diff
@@ -3,24 +3,44 @@
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import Button from "../Button-Link/Button/Button";
 import Input from "../Input/Input";
-import {
-  faEye,
-  faFileText,
-  faLock,
-  faUser,
-} from "@fortawesome/free-solid-svg-icons";
-import Link from "../Button-Link/Link/Link";
+import { faFileText, faLock, faUser } from "@fortawesome/free-solid-svg-icons";
+import { useState } from "react";
+import { registerUser } from "@/services/AuthAPI";
+import { useRouter } from "next/navigation";
 
 const Register = () => {
-  const sendRegistration: React.FormEventHandler<HTMLFormElement> = (e) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [formError, setFormError] = useState("");
+
+  const router = useRouter();
+
+  const sendRegistration: React.FormEventHandler<HTMLFormElement> = async (
+    e
+  ) => {
     e.preventDefault();
+    setFormError("");
     const formData = {
       fullName: e.currentTarget.fullName.value,
       email: e.currentTarget.email.value,
       password: e.currentTarget.password.value,
     };
-
-    console.log(formData);
+    setIsLoading(true);
+    try {
+      let resp = await registerUser(
+        formData.email,
+        formData.password,
+        formData.fullName
+      );
+      if (resp.ok === true) {
+        router.push("/");
```

Comment:
Let's use `router.replace`. The difference is:

* with `push` a new route is "added". This means the user can click "back" in browser to get to the previous route (ie: this one, the register). This doesn't make sense, since we don't show the register route for logged in users
* with `replace` this route is replaced and navigating backwards goes to the previous route